sdk/core/azure-common/azure/common/client_factory.py

```diff
@@ -74,12 +74,17 @@ def get_client_from_cli_profile(client_class, **kwargs):
     """
     cloud = get_cli_active_cloud()
     parameters = {}
-    if 'credentials' not in kwargs or 'subscription_id' not in kwargs:
+    kwarg_cred = kwargs.pop('credentials', kwargs.pop('credential', None))
```

Comment:
If (and this needs to be double-checked) we ended up specifying an explicit `None` as credentials for storage for anonymous auth, then we probably need a sentinel value other than `None` here.

Reply:
Ok, that's fair point

Reply:
Fixed